SDWebImage/Core/SDAnimatedImageView.m

```diff
@@ -501,6 +501,17 @@ - (void)displayLayer:(CALayer *)layer
     }
 }
 
+#if SD_UIKIT
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
+    // See: #3635
+    // From iOS 17, when UIImageView entering the background, it will receive the trait collection changes, and modify the CALayer.contents by `self.image.CGImage`
+    // However, For animated image, `self.image.CGImge != self.currentFrame.CGImage`, right ?
+    // So this cause the render issue, we need to reset the CALayer.contents again
+    [super traitCollectionDidChange:previousTraitCollection];
+    [self.imageViewLayer setNeedsDisplay];
```

Comment:
@longjianjiang Should we guard an `@available (iOS 17.0, *)` version check here ?

From